### PR TITLE
Migrate to pion/webrtc/v4

### PR DESCRIPTION
**Pull request by 3DRX**
#### Description
Migrate to pion/webrtc/v4, code changes:
- Change `webrtc.RtpTransceiverInit` to `webrtc.RTPTransceiverInit`.

### CODE REVIEW

**Comment by 3DRX** [comment]
@Sean-Der @at-wat Please take time to review this, huge thanks.

**Review by at-wat — COMMENTED** [review]
@3DRX thank you for your contribution!

I tried `examples/webrtc` on this branch and got:
```
Connection State has changed checking 
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
x264 [info]: profile Constrained Baseline, level 3.0, 4:2:0, 8-bit
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7c973a]

goroutine 36 [running]:
github.com/pion/mediadevices.(*baseTrack).rtcpReadLoop(0x799320?, {0x0, 0x0}, {0x7f096db2a068, 0xc0002be3f0}, 0xc0002b51f0)
	/home/at-wat/go/src/github.com/pion/mediadevices/track.go:252 +0x7a
created by github.com/pion/mediadevices.(*baseTrack).bind in goroutine 1
	/home/at-wat/go/src/github.com/pion/mediadevices/track.go:235 +0x858
exit status 2
```

Does it work on you environment?

**Comment by 3DRX** [comment]
@at-wat thanks for review! I believe this is fixed by https://github.com/pion/webrtc/commit/2553783e53d96c1e3353274e1ab41ce22c6fa06e, I'll update pion/webrtc to v4.0.5.

**Comment by 3DRX** [comment]
Yeah, this works for me.

File: examples/go.mod [hunk 1]
```diff
@@ -5,7 +5,33 @@ go 1.21
 require (
 	github.com/esimov/pigo v1.4.6
 	github.com/pion/mediadevices v0.0.0
-	github.com/pion/webrtc/v3 v3.3.4
+	github.com/pion/webrtc/v4 v4.0.2
```
Comment (at-wat) [inline examples/go.mod:8]: Need `go mod tidy` examples as well

**Review by at-wat — COMMENTED** [review]
I confirmed the example works now!
Please run `go mod tidy` under examples directory as well.

File: examples/go.mod (hunk 1 above)
Comment (3DRX) [reply examples/go.mod]: done

**Review by at-wat — APPROVED** [review]
LGTM!

**Comment by at-wat** [comment]
Thank you for your contribution!

**Pull request merged**